dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/ExpressionDimensionItemController.java

```diff
@@ -0,0 +1,200 @@
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+/**
+ * @author Dusan Bernat
+ */
+
+/**
+ *
+ */
+@Controller
+@RequestMapping( value = "/expressionDimensionItems" )
+@ApiVersion( { DhisApiVersion.DEFAULT, DhisApiVersion.ALL } )
+@AllArgsConstructor
+public class ExpressionDimensionItemController
+{
+    private final ExpressionDimensionItemService expressionService;
+
+    /**
+     *
+     * @param descriptionFilter
+     * @return
+     */
+    @ResponseBody
+    @GetMapping( produces = APPLICATION_JSON_VALUE )
+    public ResponseEntity<List<ExpressionDimensionItem>> getExpressionDimensionItems(
+        @RequestParam( required = false ) String descriptionFilter )
+    {
+        try
+        {
+            List<ExpressionDimensionItem> expressions = new ArrayList<>();
+
+            if ( descriptionFilter == null )
+            {
+                expressions.addAll( expressionService.getAllExpressionDimensionItems() );
+            }
+            else
+            {
+                expressions.addAll( expressionService.getAllExpressionDimensionItems()
+                    .stream().filter( item -> item.getDescription().contains( descriptionFilter ) )
+                    .collect( Collectors.toList() ) );
+            }
+            if ( expressions.isEmpty() )
+            {
+                return new ResponseEntity<>( HttpStatus.NO_CONTENT );
+            }
+
+            return new ResponseEntity<>( expressions, HttpStatus.OK );
+        }
+        catch ( Exception e )
+        {
+            return new ResponseEntity<>( HttpStatus.INTERNAL_SERVER_ERROR );
+        }
+    }
+
+    /**
+     *
+     * @param id
+     * @return
+     */
+    @ResponseBody
+    @GetMapping( path = "{id}", produces = APPLICATION_JSON_VALUE )
+    public ResponseEntity<ExpressionDimensionItem> getExpressionDimensionItems( @PathVariable Long id )
+    {
+        try
+        {
+            ExpressionDimensionItem item = expressionService.getExpressionDimensionItem( id );
+            return item != null ? new ResponseEntity<>( item, HttpStatus.OK )
+                : new ResponseEntity<>( HttpStatus.NO_CONTENT );
+
+        }
+        catch ( Exception e )
+        {
+            return new ResponseEntity<>( HttpStatus.INTERNAL_SERVER_ERROR );
+        }
+    }
+
+    /**
+     *
+     * @param expressionDimensionItem
+     * @return
+     */
+    @ResponseBody
+    @PostMapping( produces = APPLICATION_JSON_VALUE )
+    public ResponseEntity<ExpressionDimensionItem> createExpressionDimensionItem(
+        @RequestBody ExpressionDimensionItem expressionDimensionItem )
+    {
+        try
+        {
+            expressionService.addExpressionDimensionItem( expressionDimensionItem );
```

Comment:
is it safe to add an expressionItem without checking if another with the same id already exists?